krator/src/object.rs

```diff
@@ -24,7 +24,7 @@ impl<R: Resource> From<&R> for ObjectKey {
     fn from(object: &R) -> ObjectKey {
         ObjectKey {
             namespace: object.namespace(),
-            name: object.name(),
+            name: object.name_unchecked(),
```

Comment:
This reproduces the original behavior which can panic. The alternative is `name_any` which may return `generate_name` or empty string as well. 

This reveals some possible refactors:
* Support `generate_name`
* Anywhere that we need to uniquely track the object, use `uid`. 
* Anywhere that we need the actual name for sure should be made failable.